Tests/js/views/services/plugins/assets/ez-locationswapplugin-tests.js

```diff
@@ -0,0 +1,499 @@
+                    Assert.areEqual(
+                        e.notification.timeout, 5,
+                        "The timeout of notification should be set to 5"
+                    );
+                }
+                if (e.notification.state === 'done') {
+                    successNotificationFired = true;
+                    Assert.isTrue(
+                        (e.notification.text.indexOf(that.contentJson.name) >= 0),
+                        "The notification should contain name of content"
+                    );
+                    Assert.isTrue(
+                        (e.notification.identifier.indexOf(that.locationJson.id) >= 0),
+                        "The notification should contain id of location"
+                    );
+                    Assert.isTrue(
+                        (e.notification.text.indexOf(that.destinationContent.name) >= 0),
+                        "The notification should contain name of destination content"
+                    );
+                    Assert.areEqual(
+                        e.notification.timeout, 5,
+                        "The timeout of notification should be set to 5"
+                    );
+                }
+                if (e.notification.state === 'error') {
+                    errorNotificationFired = true;
+                }
+            });
+
+            this.service.fire('swapLocation', {location: this.location});
+
+            Mock.expect(this.app, {
+                method: 'navigateTo',
+                args: ['viewLocation', Mock.Value.Object],
+            });
+
+            Assert.isTrue(startNotificationFired, 'Should fire notification with `started` state');
+            Assert.isTrue(successNotificationFired, 'Should fire notification with `done` state');
+            Assert.isFalse(errorNotificationFired, 'Should not fire notification with `error` state');
+        },
+
+        "Should fire notifications if swap fails": function () {
+            var startNotificationFired = false,
+                successNotificationFired = false,
+                errorNotificationFired = false,
+                selection = this._getSelection(),
+                that = this;
+
+            this.service.on('contentDiscover', function (e) {
+                var config = {
+                    selection: selection,
+                };
+
+                e.config.contentDiscoveredHandler(config);
+            });
+
+            Mock.expect(this.location, {
+                'method': 'swap',
+                'args': [Mock.Value.Object, selection.location, Mock.Value.Function],
+                'run': function (options, destinationLocation, callback) {
+                    Assert.areSame(
+                        options.api,
+                        that.capi,
+                        'CAPI should be passed as param'
+                    );
+                    callback(true);
+                }
+            });
+
+            this.service.on('notify', function (e) {
```

Comment:
Is that a copy paste from the previous method, if yes could we put this in a protected method and call it in both cases.